pyresample/test/test_geometry/test_area.py

```diff
@@ -0,0 +1,2028 @@
+# Copyright (C) 2010-2020 Pyresample developers
+#
+# This program is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+# details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Test AreaDefinition objects."""
+
+import sys
+import unittest
+from unittest.mock import MagicMock, patch
+
+import dask.array as da
+import numpy as np
+import pytest
+import xarray as xr
+from pyproj import CRS, Proj
+
+from pyresample import geo_filter, parse_area_file
+from pyresample.future.geometry import AreaDefinition, BaseDefinition, SwathDefinition
+from pyresample.future.geometry.area import (
+    get_geostationary_angle_extent,
+    get_geostationary_bounding_box,
+)
+from pyresample.future.geometry.base import get_array_hashable
+from pyresample.test.utils import catch_warnings
+
+
+class Test(unittest.TestCase):
```

Comment:
I suppose all these tests are a copy of the old ones with the new definition. If the idea is to make sure the legacy functionality is preserved, could we parametrize this class instead (if that is possible?)?

Reply:
Yeah I was thinking that too. Especially when I found that some of the test methods were in the wrong class so I moved them in this new module. I'll see what I can do. Should I remove the equivalent tests in the old test module then?

Reply:
So I started down this road and hit a problem. I can make a pytest fixture that is parametrized with the area definition class to use, but the classes don't have the same arguments. The proj_id and description don't exist anymore (and depending on what we think about area_id being metadata that might go away too). This means that every use case in every test has to have different parameters depending on what class it is. I could make a helper function that is called to create the area and it ignores the parameters that have changed, but I don't really want to write the tests for the old class (using its parameters) and then eventually have to remove those arguments.

Reply:
So I caved and I did this and got it all to work...I think. I'll make a separate comment to summarize.